### ISSUE TITLE
NIF compiled as Mach format on Linux

### ISSUE THREAD

**Issue by pallix** [TattdCodeMonkey/crc#15]
Hello,

Using `crc` on Linux causes the following problem:

```
iex(1)> CRC.crc_16("hello")
2017-11-23 12:37:09.102 pid=<0.1289.0> [warn] : The on_load function for module crc_nif returned:
{:error,
 {:load_failed,
  'Failed to load NIF library: \'/home/user/projects/xyz/_build/dev/lib/crc/priv/crc_nif.so: invalid ELF header\''}}

** (UndefinedFunctionError) function :crc_nif.crc_16/1 is undefined (module :crc_nif is not available)
    (crc) :crc_nif.crc_16("hello")
```

The error is due to [crc-0.7.0 package](https://repo.hex.pm/tarballs/crc-0.7.0.tar), containing the files `c_src/env.mk` and `priv/crc_nif.so`. Since these files are built and environment dependent, they must not be included in the package. `priv/crc_nif.so` is a MACH format compiled binary, causing the error.

**Comment by TattdCodeMonkey** [comment]
There were changes to the make file in v0.7 which is the likely cause here. 

All the previous versions also had nifs and .so files too. 

I’ll see about reproducing and working on a solution to this sometime this week. Thanks for logging the issue.

Cc: @potatosalad

**Comment by potatosalad** [comment]
@TattdCodeMonkey (initial guess) It sounds like `c_src/env.mk` and the compiled shared library was included in the package published to hex.

**Comment by TattdCodeMonkey** [comment]
@pallix I just published v0.7.1 That should not include `c_src/env.mk`, `*.o` or `*.so` files.

**Comment by TattdCodeMonkey** [comment]
Tested `v0.7.1` in vagrant with ubuntu and latest elixir & erlang and everything compiled and tested 👍  closing this issue. Thanks again for reporting.
